miflora-mqtt-daemon.py

```diff
@@ -279,16 +284,22 @@ def flores_to_openhab_items(flores, reporting_mode):
                     print_line('Retrying ...', warning = True)
                 flora['poller']._cache = None
                 flora['poller']._last_read = None
+                if reporting_mode == 'mqtt-smarthome':
+                    mqtt_client.publish('{}/connected'.format(base_topic), payload='1', retain=True)
 
         if not flora['poller']._cache:
             flora['stats']['failure'] = flora['stats']['failure'] + 1
             print_line('Failed to retrieve data from Mi Flora sensor "{}" ({}), success rate: {:.0%}'.format(
                 flora['name_pretty'], flora['mac'], flora['stats']['success']/flora['stats']['count']
                 ), error = True, sd_notify = True)
             print()
+            if reporting_mode == 'mqtt-smarthome':
+                mqtt_client.publish('{}/connected'.format(base_topic), payload='1', retain=True)
             continue
         else:
             flora['stats']['success'] = flora['stats']['success'] + 1
+            if reporting_mode == 'mqtt-smarthome':
+                mqtt_client.publish('{}/connected'.format(base_topic), payload='2', retain=True)
```

Comment:
Are these three additions necessary? It's quite common for one sensor to be available only every other try. I'm even seeing a 20% success rate on one of my (deliberately) bad placed sensors.

Are these updates from 1->2->1->2 really meaningful? I'd prefer if we can leave out mode specific routines in most of the daemons parts (Which I'd like to separate into functions at some point)
In my understanding the connected state could be 0 for offline, 1 for initializing and 2 after the Initialization was successful. Your proposal defines 2 as "connected to MQTT and hardware" and I think it's fitting this way. Wdyt?

Reply:
hmm yes you're right. There is also another point: the connected topic normally indicates the successful connection to some kind of gateway. In this project we don't have the "one" gateway but multiple devices, so it doesn't really make sense. Will change that and just publish `0` and `2`.